code/modules/power/solar.dm

```diff
@@ -528,11 +528,15 @@
 	track = 2 // Auto tracking mode
 
 /obj/machinery/power/solar_control/autostart/Initialize()
-	. = ..()
-	src.search_for_connected()
+	..()
+	do_solars()
+	addtimer(CALLBACK(src, .proc/do_solars), 1800) // do it again three minutes later, just to be sure
```

Comment:
I'm pretty sure you can handle this in LateInitialize, you just need to check our wiki on how to actually do post-mapload late init.

Reply:
I'm not keen on it, I barely got this working as-is. I also kinda like that the solars only kick in some  time after the shift starts.